app/cdap/components/Connections/Browser/GenericBrowser/apiHelpers.ts

```diff
@@ -50,7 +54,11 @@ export function createWorkspace({ entity, connection, limit = 1000 }) {
       context: getCurrentNamespace(),
     },
     body
-  ).toPromise();
+  )
+    .toPromise()
+    .catch((e) => {
+      return getApiErrorMessage(e);
```

Comment:
I'm rusty on my Promises. Will returning the string from `.catch` result in a rejected Promise or a resolved Promise?

Reply:
my understanding is it will catch the error from a rejected promise, then return it so that it can hit the catch in try catch. @lahwran do you remember? I feel a bit rusty as well.

Reply:
Safer to just `return Promise.reject(getApiErrorMessage(e));`?

Reply:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/catch

Reply:
my intent was to not have to call `getErrorMessage` on each individual `catch(e) {}` block for logging purposes. I think i remembered it working this way and from a small initial test it seemed to be but i'm not 100% sure.

Reply:
Nick is correct, catch's return value becomes a resolved promise, not a rejected one. if you'd like to reject, raise a value as an error, or do what nick suggested.
<img width="512" alt="Screen Shot 2021-08-10 at 1 13 00 PM" src="https://user-images.githubusercontent.com/550498/128928585-dad9d713-da1b-496c-9d19-acd5e654de46.png">